egret/model_library/unit_commitment/security_constraints.py

```diff
@@ -0,0 +1,113 @@
+#  ___________________________________________________________________________
+#
+#  EGRET: Electrical Grid Research and Engineering Tools
+#  Copyright 2019 National Technology & Engineering Solutions of Sandia, LLC
+#  (NTESS). Under the terms of Contract DE-NA0003525 with NTESS, the U.S.
+#  Government retains certain rights in this software.
+#  This software is distributed under the Revised BSD License.
+#  ___________________________________________________________________________
+
+## for adding generic constraints to the unit commitment model
+
+from pyomo.environ import *
+from .uc_utils import add_model_attr, uc_time_helper
+from egret.data.data_utils import _recurse_into_time_index
+
+
+@add_model_attr('security_constraints', requires = {'data_loader' : None,
+                                                    'status_vars' : None,
+                                                    'power_vars' : None,
+                                                    'storage_service' : None,
+                                                    'power_balance' : None,
+                                                    })
+def security_constraint_model(model):
```

Comment:
Can you add a docstring to this function? It would be nice to know how the model is being modified.

Reply:
Done!